packages/itwin/ec3-widget/src/components/CreateAssemblyComponent.tsx

```diff
@@ -0,0 +1,333 @@
+                        />
+                        <LabeledSelect
+                          data-testid="ec3-element-quantity-select"
+                          required
+                          label={"Element quantity"}
+                          options={
+                            getMetadataColumns(assembly, CreateAssemblyDropdownType.elementQuantity)?.map((x) => {
+                              return { label: x, value: x };
+                            }) ?? []
+                          }
+                          value={assembly.elementQuantityColumn}
+                          onChange={async (value) => {
+                            onAssemblyDataChange({ ...assembly, elementQuantityColumn: value }, i);
+                          }}
+                          disabled={isLoading || assembly.reportTable === "" || i !== editableAssemblyIndex}
+                          placeholder={isLoading ? "Loading elements" : assembly.reportTable === "" ? "Select report table first" : "Select element quantity"}
+                        />
+                        <Label htmlFor="combo-input" required>
+                          Materials
+                        </Label>
+                        <Select
+                          disabled={isLoading || assembly.reportTable === "" || i !== editableAssemblyIndex}
+                          options={
+                            getMetadataColumns(assembly, CreateAssemblyDropdownType.material)?.map((x) => {
+                              return { label: x, value: x };
+                            }) ?? []
+                          }
+                          value={assembly.materials.map((x) => x?.nameColumn)}
+                          onChange={(val, event) => {
+                            if (val) {
+                              onAssemblyDataChange(
+                                {
+                                  ...assembly,
+                                  materials:
+                                    event === "removed"
+                                      ? assembly.materials.filter((value) => val !== value?.nameColumn)
+                                      : [...(assembly.materials ?? []), { nameColumn: val }],
+                                },
+                                i,
+                              );
+                            }
+                          }}
+                          placeholder={
+                            isLoading
+                              ? "Loading elements"
+                              : assembly.reportTable === ""
+                                ? "Select report table first"
+                                : "Select property containing material names"
+                          }
+                          multiple
+                        />
+                      </>
+                    </ExpandableBlock>
```

Comment:
This looks like it can be made into its own component.

Reply:
moved to a new component called `AssemblyItem`